src/main/java/pageclasses/WebDriverSingleton.java

```diff
@@ -0,0 +1,33 @@
+package pageclasses;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+import java.time.Duration;
+
+public class WebDriverSingleton {
+    private static volatile WebDriverSingleton webDriverSingleton; // volatile so it's always up-to-date
+    private final WebDriver driver;
```

Comment:
it is better to use ThreadLocal to ensure thread safety
https://medium.com/10-minutes-qa-story/run-selenium-tests-in-parallel-with-singletone-driver-21ed058c6fad